### PR TITLE
Added alarms for lambda to cloudwatch

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

Add alerts for when 500 errors appear in the API lambda logs

# Test instructions | Instructions pour tester la modification

ran terragrunt tf-plan

### CODE REVIEW

**Review by sastels — APPROVED** [review]
LGTM!

File: env/staging/lambda-api/terragrunt.hcl [hunk 1]
```diff
@@ -15,7 +15,9 @@ dependency "common" {
       "",
       "",
     ]
-    sns_alert_general_arn = ""
+    sns_alert_general_arn = "",
+    sns_alert_warning_arn = "",
```
Comment (patheard) [inline env/staging/lambda-api/terragrunt.hcl:19]: You've got some trailing `,` that can be removed here.

File: aws/lambda-api/cloudwatch_alarms.tf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  alarm_actions       = [var.sns_alert_warning_arn]
+}
+
+resource "aws_cloudwatch_metric_alarm" "logs-10-500-error-5-minutes-critical-lambda-api" {
+  alarm_name          = "logs-10-500-error-5-minutes-critical-lambda-api"
+  alarm_description   = "Ten 500 errors in 5 minutes for lambda api"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "1"
+  metric_name         = aws_cloudwatch_log_metric_filter.web-500-errors-lambda-api.metric_transformation[0].name
+  namespace           = aws_cloudwatch_log_metric_filter.web-500-errors-lambda-api.metric_transformation[0].namespace
```
Comment (patheard) [inline aws/lambda-api/cloudwatch_alarms.tf:27]: Ah nice!  I like this as a way to reference the metric filter - much cleaner than what I've done in the past.

File: aws/lambda-api/cloudwatch_alarms.tf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  alarm_description   = "One 500 error in 1 minute for lambda api"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "1"
+  metric_name         = aws_cloudwatch_log_metric_filter.web-500-errors-lambda-api.metric_transformation[0].name
+  namespace           = aws_cloudwatch_log_metric_filter.web-500-errors-lambda-api.metric_transformation[0].namespace
+  period              = "60"
+  statistic           = "Sum"
+  threshold           = 1
+  treat_missing_data  = "notBreaching"
+  alarm_actions       = [var.sns_alert_warning_arn]
```
Comment (patheard) [inline aws/lambda-api/cloudwatch_alarms.tf:18]: Do you want an `ok_actions` here as well, like what you have in the `aws_cloudwatch_metric_alarm.logs-10-500-error-5-minutes-critical-lambda-api` below?  Personally I find them helpful.

**Review by patheard — CHANGES_REQUESTED** [review]
A few minor comments/questions, but looks good 🎉

File: aws/lambda-api/cloudwatch_alarms.tf (hunk 3 above)
Comment (jimleroyer) [reply aws/lambda-api/cloudwatch_alarms.tf:18]: That could be useful, and we could add it to the missing ones such as `logs-1-500-error-1-minute-warning`.

**Review by patheard — APPROVED** [review]

**Pull request merged**